src/bio/bio_recovery.c

```diff
@@ -236,18 +442,14 @@ bio_bs_state_set(struct bio_blobstore *bbs, enum bio_bs_state new_state)
 			bio_state_enum_to_str(new_state));
 		bbs->bb_state = new_state;
 
-		if (new_state == BIO_BS_STATE_NORMAL ||
-		    new_state == BIO_BS_STATE_FAULTY) {
+		if (new_state == BIO_BS_STATE_FAULTY) {
 			struct spdk_bs_type	bstype;
 			uuid_t			dev_id;
-			enum smd_dev_state	dev_state;
 
 			bstype = spdk_bs_get_bstype(bbs->bb_bs);
 			memcpy(dev_id, bstype.bstype, sizeof(dev_id));
-			dev_state = new_state == BIO_BS_STATE_NORMAL ?
-					SMD_DEV_NORMAL : SMD_DEV_FAULTY;
 
-			rc = smd_dev_set_state(dev_id, dev_state);
+			rc = smd_dev_set_state(dev_id, SMD_DEV_FAULTY);
```

Comment:
Right, default SMD device state is NORMAL. It can be changed to FAULTY through control plane set-faulty command (or by auto faulty detection). The FAULTY state can be reverted back to NORMAL by device replace command, I'll handle device replace in follow-on patch.